yb-voyager/src/queryparser/unsupported_constructs.go

```diff
@@ -0,0 +1,121 @@
+package queryparser
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	ADVISORY_LOCKS = "Advisory Locks"
+	SYSTEM_COLUMNS = "System Columns"
+	XML_FUNCTIONS  = "XML Functions"
+)
+
+// To Add a new unsupported query construct implement this interface
+type UnsupportedConstructDetector interface {
+	Detect(msg protoreflect.Message) ([]string, error)
+}
+
+type FuncCallDetector struct {
+	// right now it covers Advisory Locks and XML functions
+	unsupportedFuncs map[string]string
+}
+
+func NewFuncCallDetector() *FuncCallDetector {
+	fcd := &FuncCallDetector{
+		unsupportedFuncs: make(map[string]string),
+	}
+
+	for _, fname := range unsupportedAdvLockFuncs {
+		fcd.unsupportedFuncs[fname] = ADVISORY_LOCKS
+	}
+	for _, fname := range xmlFunctions {
+		fcd.unsupportedFuncs[fname] = XML_FUNCTIONS
+	}
+	log.Debug("initialized FuncCallDetector with unsupported functions")
+	return fcd
+}
+
+// Detect checks if a FuncCall node uses an unsupported function.
+func (d *FuncCallDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if getMsgFullName(msg) != PG_QUERY_FUNCCALL_NODE {
+		return nil, nil
+	}
+
+	funcName := getFuncNameFromFuncCall(msg)
+	log.Debugf("fetched function name from %s node: %q", PG_QUERY_FUNCCALL_NODE, funcName)
+	if constructType, ok := d.unsupportedFuncs[funcName]; ok {
+		log.Debugf("detected unsupported function: %s", funcName)
```

Comment:
Consider also logging the msg? . Otherwise, it might not be useful. Same for the other types of detection logs

Reply:
`msg` variable directly wont be of any use since it has pointers also.
I was thinking `msg.Descriptor()` but check it for a case and didn't see anything helpful in that either

```
func call name "pg_advisory_lock" in msg MessageDescriptor{
	Syntax:   proto3
	FullName: pg_query.FuncCall
	Fields: [{
		Name:        funcname
		Number:      1
		Cardinality: repeated
		Kind:        message
		HasJSONName: true
		JSONName:    "funcname"
		IsList:      true
		Message:     pg_query.Node
	}, {
		Name:        args
		Number:      2
		Cardinality: repeated
		Kind:        message
		HasJSONName: true
		JSONName:    "args"
		IsList:      true
		Message:     pg_query.Node
	}, {
		Name:        agg_order
		Number:      3
		Cardinality: repeated
		Kind:        message
		HasJSONName: true
		JSONName:    "agg_order"
		IsList:      true
		Message:     pg_query.Node
	}, {
		Name:        agg_filter
		Number:      4
		Cardinality: optional
		Kind:        message
		HasJSONName: true
		JSONName:    "agg_filter"
		HasPresence: true
		Message:     pg_query.Node
	}, {
		Name:        over
		Number:      5
		Cardinality: optional
		Kind:        message
		HasJSONName: true
		JSONName:    "over"
		HasPresence: true
		Message:     pg_query.WindowDef
	}, {
		Name:        agg_within_group
		Number:      6
		Cardinality: optional
		Kind:        bool
		HasJSONName: true
		JSONName:    "agg_within_group"
	}, {
		Name:        agg_star
		Number:      7
		Cardinality: optional
		Kind:        bool
		HasJSONName: true
		JSONName:    "agg_star"
	}, {
		Name:        agg_distinct
		Number:      8
		Cardinality: optional
		Kind:        bool
		HasJSONName: true
		JSONName:    "agg_distinct"
	}, {
		Name:        func_variadic
		Number:      9
		Cardinality: optional
		Kind:        bool
		HasJSONName: true
		JSONName:    "func_variadic"
	}, {
		Name:        funcformat
		Number:      10
		Cardinality: optional
		Kind:        enum
		HasJSONName: true
		JSONName:    "funcformat"
		Enum:        pg_query.CoercionForm
	}, {
		Name:        location
		Number:      11
		Cardinality: optional
		Kind:        int32
		HasJSONName: true
		JSONName:    "location"
	}]
}
```